api/src/main/java/com/community/api/endpoint/avisoft/controller/Qualification/QualificationDetailsController.java

```diff
@@ -51,39 +66,57 @@ public ResponseEntity<?> addQualificationDetail(@PathVariable Long customCustome
         {
+            exceptionHandling.handleException(e);
             return ResponseService.generateErrorResponse("Something went wrong",HttpStatus.BAD_REQUEST);
         }
     }
 
     @GetMapping("/get-by-customer-id/{customCustomerId}")
-    public ResponseEntity<?> getQualificationDetailsById(@PathVariable Long customCustomerId) throws CustomerDoesNotExistsException ,EntityDoesNotExistsException{
+    public ResponseEntity<?> getQualificationDetailsById(@PathVariable Long customCustomerId, @RequestHeader(value = "Authorization") String authHeader) throws CustomerDoesNotExistsException ,EntityDoesNotExistsException{
+        String role=null;
         try
         {
-            List<QualificationDetails> qualificationDetails = qualificationDetailsService.getQualificationDetailsByCustomerId(customCustomerId);
+            String jwtToken = authHeader.substring(7);
+            Integer roleId = jwtTokenUtil.extractRoleId(jwtToken);
+             role = roleService.getRoleByRoleId(roleId).getRole_name();
+            List<Map<String, Object>> qualificationDetails = qualificationDetailsService.getQualificationDetailsByCustomerId(customCustomerId,role);
             if(qualificationDetails.isEmpty())
             {
-                return ResponseService.generateSuccessResponse("Qualification Details list is empty ",qualificationDetails,HttpStatus.OK);
+                return ResponseService.generateSuccessResponse("Qualification Details list is empty for "+role,qualificationDetails,HttpStatus.OK);
             }
-            return ResponseService.generateSuccessResponse("Qualification Details are found ",qualificationDetails,HttpStatus.OK);
+            return ResponseService.generateSuccessResponse("Qualification Details are found for "+role,qualificationDetails,HttpStatus.OK);
         }
         catch (CustomerDoesNotExistsException e)
         {
+            if(role.equalsIgnoreCase(Constant.SERVICE_PROVIDER))
+            {
+                return ResponseService.generateErrorResponse("Service Provider does not exist",HttpStatus.NOT_FOUND);
+            }
             return ResponseService.generateErrorResponse("Customer does not exist",HttpStatus.NOT_FOUND);
         } catch (Exception e)
         {
+            exceptionHandling.handleException(e);
             return ResponseService.generateErrorResponse("Something went wrong",HttpStatus.BAD_REQUEST);
         }
     }
 
     @DeleteMapping("/delete/{customCustomerId}/{qualificationDetailId}")
-    public ResponseEntity<?> deleteQualificationDetailById(@PathVariable Long customCustomerId, @PathVariable Long qualificationDetailId) throws EntityDoesNotExistsException, CustomerDoesNotExistsException {
+    public ResponseEntity<?> deleteQualificationDetailById(@PathVariable Long customCustomerId, @PathVariable Long qualificationDetailId,@RequestHeader(value = "Authorization") String authHeader) throws EntityDoesNotExistsException, CustomerDoesNotExistsException {
```

Comment:
here if someone enters string
give NumberFormatException exception

use this
Long id = Long.valueOf(customerId);

catch (NumberFormatException e) {
    return ResponseService.generateErrorResponse("Invalid customerId: expected a Long", HttpStatus.BAD_REQUEST);
}